packages/functionals/botpress-qna/src/index.js

```diff
@@ -70,5 +74,67 @@ module.exports = {
         res.status(500).send(e.message || 'Error')
       }
     })
+
+    router.get('/csv', async (req, res) => {
+      const questions = await storage.getQuestions()
+      const rows = questions.flatMap(item => {
+        console.log(item)
+        const { questions, answer, action, redirectFlow, redirectNode } = item.data
+        return questions.map(question => ({
+          question,
+          action,
+          answer: action === 'text' ? answer : [redirectFlow, redirectNode].filter(Boolean).join('#')
+        }))
+      })
+      res.setHeader('Content-Type', 'text/csv')
+      res.setHeader('Content-disposition', `attachment; filename=qna_${moment().format('DD-MM-YYYY')}.csv`)
+      rows.forEach(({ question, action, answer }) => {
+        res.write(
+          [question, action, answer].map(field => '"' + field.toString().replace(/\"/g, '""') + '"').toString() + '\r\n'
```

Comment:
This looks fine but maybe it's better to use some external CSV formatter that will handle the escaping for us?

Reply:
Done. Used `json2csv`.